File: app/src/main/java/com/openclassrooms/realestatemanager/RealEstateRepository.java
package com.openclassrooms.realestatemanager;

import androidx.lifecycle.LiveData;

import java.util.List;

public class RealEstateRepository {

    private final RealEstateDao realEstateDao;

    public RealEstateRepository(RealEstateDao realEstateDao) {
        this.realEstateDao = realEstateDao;
    }

    public LiveData<List<RealEstate>> getAllRealEstate() { return this.realEstateDao.getAllRealEstate(); }
    public LiveData<RealEstate> getRealEstate(long realEstateId) { return this.realEstateDao.getRealEstate(realEstateId); }

    void insert(RealEstate realEstate) { realEstateDao.createRealEstate(realEstate); }
    void update(RealEstate realEstate) { realEstateDao.updateRealEstate(realEstate); }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/repository/RealEstateRepository.kt
package com.openclassrooms.realestatemanager.repository

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.openclassrooms.realestatemanager.dao.RealEstateDao
import com.openclassrooms.realestatemanager.model.RealEstate

class RealEstateRepository(private val realEstateDao: RealEstateDao) {

    private val realEstates = mutableListOf<RealEstate>()

    private val observableRealEstates = MutableLiveData<List<RealEstate>>(realEstates)

    fun getAllRealEstates(): LiveData<List<RealEstate>> = realEstateDao.getRealEstates()

    fun getRealEstate(id: Int) = realEstateDao.getRealEstate(id)

    suspend fun insertRealEstate(realEstate: RealEstate) { realEstateDao.insertRealEstate(realEstate) }

    suspend fun updateRealEstate(realEstate: RealEstate) { realEstateDao.updateRealEstate(realEstate) }

}